accepted/00000-snap-package-support.md

```diff
@@ -0,0 +1,175 @@
+             │
+             │ Note: Snap packages do not have a public URL,
+             │       so no URL is required in CLI.
+             ▼
+┌────────────────────────────────────────────┐
+│            reposync.py (main logic)        │
+│ - Selects plugin by repo_type              │
+└────────────┬───────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────┐
+│          snap_src.py (Snap plugin)         │
+│ - Creates ContentSource                    │
+│ - Wraps SnapRepo                           │
+└────────────┬───────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────────────────────┐
+│        SnapRepo.get_all_package_list()                     │
+│  - Calls: https://api.snapcraft.io/api/v1/snaps/search?q={query}&limit={limit}    │
+│  - Creates SnapPackage objects (name, path, etc.)          │
+└────────────┬───────────────────────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────────────────────┐
+│        SnapRepo.get_select_package_list()                  │
+│  - Filters selected packages                               │
+│  - Creates SnapPackage objects (name, path, etc.)          │
+└────────────┬───────────────────────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────────────────────┐
+│            SnapRepo._download(url)                         │
+│  - Performs HTTP download of selected .snap file           │
+│  - GET https://api.snapcraft.io/v2/snaps/info/<snap_name> │
+└────────────┬───────────────────────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────────────────────┐
+│      reposync.py → import_packages()                       │
+│  - Imports metadata into the database                      │
+│  - Saves .snap files under /var/spacewalk/packages/        │
+└────────────┬───────────────────────────────────────────────┘
+             │
+             ▼
+┌────────────────────────────────────────────────────────────┐
+│   Minions can now use Uyuni as a Snap repository           │
+│   (served through Uyuni-hosted channels)                   │
+│   e.g. `sudo snap install <pkg>` from Uyuni server         │
+└────────────────────────────────────────────────────────────┘
+```
+
+Fake URL Design for Snap Repositories
+
+Snap packages are distributed via the Snap Store using dynamic APIs rather than static URLs. To maintain consistency with other repository types (e.g., YUM, DEB),  introduce a Fake URL format to represent Snap sources uniformly in the database.
+
+Fake URL Format:
+```
+snap://<repo label>/<stable/edge/candidate/beta>
+```
+
+
+These URLs are not used for actual downloads. Instead, they serve as unique identifiers stored in the rhnContentSource.source_url field. The real package fetching will be handled through the Snap API.
```

Comment:
I don't think we should have a fake URL on that table. We probably need a new set of tables to specify the API from where to download (and maybe authentication data, if provided)